src/http/api/resources/files-regular.js

```diff
@@ -336,6 +336,10 @@ exports.refs = {
     const unique = request.query.unique
     const maxDepth = request.query['max-depth']
 
+    if (maxDepth <= 0) {
+      return h.response()
+    }
```

Comment:
This is because `ndjson.serialize` writes a single `\n` into the response even if the stream was empty - in the client we strip the `\n` from response and try to parse what's left as JSON which then explodes, so if we know we're not going to send anything back, shortcut the streaming of refs.

This is noted here and not in the code because it's all going to be removed when #2517 lands.